include/zephyr/acpi/acpi.h

```diff
@@ -173,6 +173,11 @@ int acpi_dmar_entry_get(enum AcpiDmarType type, ACPI_SUBTABLE_HEADER **tables);
 int acpi_drhd_get(enum AcpiDmarScopeType scope, ACPI_DMAR_DEVICE_SCOPE *dev_scope,
 		  union acpi_dmar_id *dmar_id, int *num_inst, int max_inst);
 
+typedef void (*dmar_foreach_func_t)(void *subtable);
+
+void acpi_dmar_foreach_subtable(ACPI_TABLE_DMAR *dmar, dmar_foreach_func_t func);
+void acpi_dmar_foreach_devscope(ACPI_DMAR_HARDWARE_UNIT *hu, dmar_foreach_func_t func);
```

Comment:
Does it really make sense to use the same callback type for both foreach functions? If you had a dedicated callback type, couldn't you then have the correct parameter type without the need for implicit casting from `void *`? I.e. one callback would have the input parameter type `ACPI_DMAR_DEVICE_SCOPE *` and the other would have `ACPI_DMAR_HEADER *`? Feel free to disagree - it's possible there's something I don't understand wrt to the ACPI typing here.

Reply:
Using different callbacks, probably, makes code clearer, since I do not need to cast. I used the same as I though those structures have the same headers.